lib/src/rules/sort_unnamed_constructors_first.dart

```diff
@@ -54,10 +55,9 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   _Visitor(this.rule);
 
-  @override
-  void visitClassDeclaration(ClassDeclaration node) {
+  void check(NodeList<ClassMember> classMembers) {
     // Sort members by offset.
-    var members = node.members.toList()
+    var members = classMembers.toList()
       ..sort((ClassMember m1, ClassMember m2) => m1.offset - m2.offset);
```

Comment:
I didn't think of this on the last PR, but I'm kind of surprised that we need to sort the members. I thought we already returned them in source order. Not critical, but it would improve performance to not make a pass confirming that everything is already sorted.

Reply:
Interesting!  I've made a note to loop back and check on this.  Thanks!

Reply:
Do you think it may be interesting to use a `SplayTreeSet` instead to keep always the order within the data structure?